### PR TITLE
Update api to be more extensible

### PR DESCRIPTION
**Pull request by ryanmoelter**
When I was building the sample app, I realized how difficult it was to add a new property for animation using the existing library.

This was mostly due to the typing in the `AnimationBuilder` and `AnimatorBuilder.target(*)` methods. Together, they made it so that you can only animate classes that have a specific AnimationBuilder (or you could try to get a `GenericAnimationBuilder`, but that was annoying and couldn't be combined with view-specific animations).

Overall, it was annoying. So I refactored the animation part of the dsl to be based on extension functions. This has some important trade-offs:
* DSL is harder/unreasonable to use in Java (rather than just being annoying before).
* It's slightly harder to find in autocomplete (it's prioritized below the current object's methods, which is particularly annoying when animating within views, where e.g. `view.x` will complete above the animation `x(...)`). They still show up right after the view's method, and in `ctrl+space` if you scroll a little.
  * We made a similar trade-off when deciding on a DSL over a fluent interface. We figured that people generally know what properties they want to animate going into to it.

But it has important benefits, too:
* Adding a new property to animate is as easy as creating the property and adding an extension function on `AnimationBuilder<View>`.
* Externally-defined animations/properties are semantically equivalent to built-in ones.

I think this is a trade-off worth making, as I think animating arbitrary properties is more important than being able to see all available methods easily (since those methods **will** still show up if you search for them).

TL;DR this works now, it didn't before:
```kotlin
blend {
  target(myMaterialCardView).animations {
    fadeIn()
    customProperty(Color.GREEN, strokeProperty)
  }
}.start()

...

val strokeProperty = makeAdditiveViewProperty<MaterialCardView>(
  R.id.demoCircle,
  get = { subject -> subject.strokeColorStateList!!.defaultColor.toFloat() },
  set = { subject, value -> subject.strokeColor = value.toInt() },
  interpolateAction = { startValue, endValue, timeFraction, _ ->
    blendColor(startValue, endValue, timeFraction)
  }
)
```

In this PR:
* Use extension functions in animation DSL for more flexibility
* Expose makeAdditiveViewProperty helper function
* Rename `genericProperty` -> `customProperty`

### CODE REVIEW

File: blend-library/src/main/java/com/wealthfront/blend/dsl/AnimatorBuilder.kt [hunk 1]
```diff
@@ -33,52 +33,8 @@ class AnimatorBuilder(
...
-
-  /**
-   * Target [ProgressBar]s for animation.
-   */
-  @JvmName("targetProgressBars")
-  fun target(views: List<ProgressBar>) = Targetable(
-      ProgressBarAnimationBuilder(
-          views,
-          currentAnimator))
+  fun <T : Any> target(subjects: List<T>) =
```
Comment (ryanmoelter) [inline blend-library/src/main/java/com/wealthfront/blend/dsl/AnimatorBuilder.kt:36]: We only need the one type of `target` now (this one and one above that uses varargs)

File: blend-library/src/main/java/com/wealthfront/blend/properties/AdditiveProperty.kt [hunk 2]
```diff
@@ -104,3 +105,11 @@ interface AdditiveIntProperty<in Subject> : AdditiveProperty<Subject> {
       .toFloat()
   }
 }
+
+/**
+ * An [AdditiveProperty] specifically designed for views. Handles storing running animations for you.
+ */
+interface AdditiveViewProperty<in Subject : View> : AdditiveProperty<Subject> {
```
Comment (ryanmoelter) [inline blend-library/src/main/java/com/wealthfront/blend/properties/AdditiveProperty.kt:112]: Added this because it makes more sense than keeping this logic in the helper function

File: blend-sample-app/src/main/java/com/wealthfront/blend/sample/AnimationDemoView.kt [hunk 3]
```diff
@@ -69,16 +72,34 @@ class AnimationDemoView @JvmOverloads constructor(
...
+          customProperty(Color.GREEN.toFloat(), strokeProperty)
+        } else {
+          y(bottomY)
+          customProperty(getColor(context, R.color.colorPrimary).toFloat(), strokeProperty)
+        }
       }
+    }.start()
+  }
+
+  val strokeProperty = makeAdditiveViewProperty<MaterialCardView>(
```
Comment (ryanmoelter) [inline blend-sample-app/src/main/java/com/wealthfront/blend/sample/AnimationDemoView.kt:95]: Added a custom property example to the demo
